examples/chapter1/annexb/include/simple_event_publisher.h

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2023 Klepsydra Technologies AG
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef SIMPLE_PUBLISHER_H
+#define SIMPLE_PUBLISHER_H
+
+#include <iostream>
+#include <klepsydra/core/publisher.h>
+
+#include "test_event.h"
+
+class SimpleEventPublisher
+{
+public:
+    explicit SimpleEventPublisher(kpsr::Publisher<TestEvent> *publisher)
+        : _publisher(publisher)
+    {}
+
+    void run(TestEvent event)
```

Comment:
`run(TestEvent event)` means the function is making a copy of the `event`. Instead use `run(const TestEvent &event)` since the internal of run is not going to modify `event`.

Reply:
right, and it duplicates the `copyInvocations`